sdk/core/azure-core/test/ut/context.cpp

```diff
@@ -52,7 +55,8 @@ TEST(Context, BasicStdString)
 
   // New context from previous
   auto c2 = context.WithValue(key, s);
-  auto& value = c2.GetValue<std::string>(key);
+  std::string value;
+  EXPECT_TRUE(c2.TryGetValue<std::string>(key, value));
   EXPECT_TRUE(value == s);
```

Comment:
nit: if you cound update `EXPECT_TRUE(x == y)` to `EXPECT_EQ(x, y)`, that'd be great. Logs/test failure messages are more informative this way.

Reply:
Unrelated to this PR and is an existing issue throughout the SDK tests. Leaving as is for now. https://github.com/Azure/azure-sdk-for-cpp/issues/2057